File: server/BeerWorkshop.Application/Dto/Requests/ProductsInventory/CreateProductInventoryItemRequestDto.cs
using BeerWorkshop.Database.Enums;

namespace BeerWorkshop.Application.Dto.Requests.ProductsInventory;

public record CreateProductInventoryItemRequestDto(
    Guid ProductId,
    decimal Quantity,
    DateTime IncomingDate,
    DateTime? OpeningDate,
    decimal PurchasePrice,
    decimal RetailPrice,
    decimal PricePerQuantity,
    DateTime ManufactureDate,
    int ExpirationTime,
    ExpirationMeasure ExpirationMeasure,
    ExpirationCountingDateType ExpirationCountingDateType
);

File: client/src/redux/dtos/requests/productsInventory/createProductInventoryItemRequestDto.ts
import {
  ExpirationCountingDateType,
  ExpirationMeasureType,
} from "../../../enums";

export interface CreateProductInventoryItemRequestDto {
  quantity: number;
  incomingDate: Date;
  purchasePrice: number;
  purchaseVat: number;
  retailPrice: number;
  pricePerQuantity: number;
  manufactureDate: Date;
  expirationTime: number;
  expirationMeasure: ExpirationMeasureType;
  openingDate?: Date | null;
  expirationCountingDateType: ExpirationCountingDateType;
  productId: string;
}
